nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -2195,13 +2195,16 @@ private boolean mayBeNullExpr(VisitorState state, ExpressionTree expr) {
       return false;
     }
     // the logic here is to avoid doing dataflow analysis whenever possible
-    Symbol exprSymbol = ASTHelpers.getSymbol(expr);
-    boolean exprMayBeNull;
     switch (expr.getKind()) {
       case NULL_LITERAL:
         // obviously null
-        exprMayBeNull = true;
+        return true;
+      default:
         break;
+    }
+    Symbol exprSymbol = ASTHelpers.getSymbol(expr);
+    boolean exprMayBeNull;
+    switch (expr.getKind()) {
       case ARRAY_ACCESS:
         // unsound!  we cannot check for nullness of array contents yet
         exprMayBeNull = false;
```

Comment:
imo this is another candidate to be pulled into the 1st switch to `return false` immediately?
i might be missing something but what value is there in passing this through the handlers chain when none of them deal with ARRAY_ACCESS expressions?
this can then of course be revisited once there is a dedicated handler for this.

same imo for the expression types in the next block that are `// clearly not null`

Reply:
For array accesses I'd rather not bail out for now.  For array accesses we are thinking about doing better handling according to JSpecify in the medium term, and it's possible a handler may want to override what happens there.

For the `// clearly not null` cases I'm not sure.  @lazaroclapp do you think it's worth bailing out early to avoid running the handlers for these cases, where basically according to the language spec the expression cannot be null?

Reply:
imo as long as no handler cares for these expression it is best to return early, because this avoids redundant work (calling the handler chain for no reason).
see also suggestion in https://github.com/uber/NullAway/pull/753

if in the near future a handler for array access will be added, it is trivial to move that expression back to the 2nd switch ...
if proper tests get added this required move will become obvious.